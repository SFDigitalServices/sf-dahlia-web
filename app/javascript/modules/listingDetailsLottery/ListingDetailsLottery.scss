.modal {
  @apply max-w-0;
}

.modal__inner {
  @apply p-0;
}

.modal__close {
  z-index: 100;
}
