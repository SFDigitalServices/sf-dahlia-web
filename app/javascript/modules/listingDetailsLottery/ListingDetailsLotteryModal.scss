.lottery-modal {
    .lottery-results-form .input {
        border-radius: initial
    }
}