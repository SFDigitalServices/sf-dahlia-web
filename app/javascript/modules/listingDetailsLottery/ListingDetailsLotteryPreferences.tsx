import React from "react"
import { t } from "@bloom-housing/ui-components"
import dayjs from "dayjs"
import { RailsLotteryBucketsDetails } from "../../api/types/rails/listings/RailsLotteryBucketsDetails"
import { RailsListing } from "../listings/SharedHelpers"

export interface ListingDetailsLotteryPreferencesProps {
  listing: RailsListing
  lotteryBucketsDetails: RailsLotteryBucketsDetails
}

export const ListingDetailsLotteryPreferences = ({
  listing,
  lotteryBucketsDetails,
}: ListingDetailsLotteryPreferencesProps) => {
  return (
    <div className="text-tiny">
      <p className="py-4 text-sm">{t("lottery.buckets_intro")}</p>
      <h2 className="text-caps-underline">{t("lottery.housing_preferences")}</h2>
      <p className="border-b border-gray-450 mb-4 pb-4 text-gray-700 text-sm">
        {t("lottery.ranking_order_note")}
      </p>
      {lotteryBucketsDetails.lotteryBuckets
        .filter((bucket) => bucket.preferenceName !== "generalLottery")
        .map((bucket, index, arr) => (
          <>
            <h3
              className="font-sans font-semibold text-tiny tracking-wide uppercase"
              key={bucket.preferenceOrder}
            >
              {bucket.preferenceName}
            </h3>
            <p className="text-tiny">
              {t("lottery.up_to_x_units_available", { units: bucket.unitsAvailable })}
            </p>
            <p className="text-gray-700 text-tiny">
              {t("lottery.number_applicants_qualified_for_preference", {
                number: bucket.totalSubmittedApps,
              })}
            </p>
            <hr className={arr.length - 1 === index ? "mt-4" : "my-4"} />
          </>
        ))}
      <div className="bg-gray-100 border-b mb-4 p-4">
        <hr className="border-b-4 border-primary" />
      </div>
      {lotteryBucketsDetails.lotteryBuckets
        .filter((bucket) => bucket.preferenceName === "generalLottery")
        .map((bucket) => (
          <div key={bucket.preferenceOrder}>
            <h3 className="font-sans font-semibold text-tiny tracking-wide uppercase">
              {t("lottery.general_pool")}
            </h3>
            <p className="text-tiny">{t("lottery.any_remaining_units")}</p>
            <p className=" border-b pb-2 text-gray-700 text-tiny">
              {t("lottery.number_applicants_qualified_for_general_pool", {
                number: bucket.totalSubmittedApps,
              })}
            </p>
          </div>
        ))}
      <div className="bg-gray-100 mb-2 py-4 text-center">
        <a href="http://sfmohcd.org/housing-preference-programs" target="_blank">
          {t("lottery.read_about_preferences")}
        </a>
      </div>
      <div className="text-center">
        <p className="py-2">
          {t("lottery.results_from", {
            date: dayjs(listing.Lottery_Results_Date).format("MMM D, YYYY"),
          })}
        </p>
        <p className="py-2">
          <a href={listing.LotteryResultsURL} target="_blank">
            {t("lottery.lottery_results_pdf_link")}
          </a>
        </p>
        <p className="text-gray-700 py-2">
          {t("lottery.unsorted_results_note", {
            date: dayjs(listing.Lottery_Date).format("MMM D, YYYY"),
          })}
        </p>
      </div>
    </div>
  )
}
