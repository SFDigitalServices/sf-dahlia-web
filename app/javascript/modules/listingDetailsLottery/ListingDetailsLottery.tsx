import React, { useEffect, useState } from "react"
import {
  AppearanceSizeType,
  AppearanceStyleType,
  Button,
  Heading,
  Modal,
  t,
} from "@bloom-housing/ui-components"
import { RailsListing } from "../listings/SharedHelpers"
import { isLotteryComplete } from "../../util/listingUtil"
import { getLotteryBucketDetails } from "../../api/listingApiService"
import { RailsLotteryResult } from "../../api/types/rails/listings/RailsLotteryResult"
import { ListingDetailsLotteryModal } from "./ListingDetailsLotteryModal"
import { localizedFormat, renderInlineWithInnerHTML } from "../../util/languageUtil"

export interface ListingDetailsLotteryProps {
  listing: RailsListing
}

export const ListingDetailsLottery = ({ listing }: ListingDetailsLotteryProps) => {
  const [lotteryBucketDetails, setLotteryBucketDetails] = useState<RailsLotteryResult>()
  const [isModalOpen, setIsModalOpen] = useState(false)

  useEffect(() => {
    void getLotteryBucketDetails(listing.Id).then((lotteryBucketDetails) => {
      setLotteryBucketDetails(lotteryBucketDetails)
    })
  }, [listing.Id])

  return (
    isLotteryComplete(listing) && (
      <div className="border-b pt-4 text-center">
        <Heading className="mb-4" priority={4}>
          {t("lottery.lotteryResults")}
        </Heading>
        <p className="mb-4 text-sm uppercase">
          {localizedFormat(listing.Lottery_Results_Date, "LL")}
        </p>
        <div className="bg-gray-100 py-4">
          {listing.Lottery_Summary && (
            <div className="mb-3 mx-2 text-gray-700 text-sm">
              {renderInlineWithInnerHTML(listing.Lottery_Summary)}
            </div>
          )}
          <Button
            size={AppearanceSizeType.small}
            styleType={AppearanceStyleType.primary}
            onClick={() => {
              setIsModalOpen(true)
            }}
          >
            {t("listings.lottery.viewLotteryResults")}
          </Button>
        </div>

        {lotteryBucketDetails && (
          <Modal onClose={() => setIsModalOpen(false)} open={isModalOpen} title="">
            <ListingDetailsLotteryModal
              listing={listing}
              lotteryBucketDetails={lotteryBucketDetails}
            />
          </Modal>
        )}
      </div>
    )
  )
}
