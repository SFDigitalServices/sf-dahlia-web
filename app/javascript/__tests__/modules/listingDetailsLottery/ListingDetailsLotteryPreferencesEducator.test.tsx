import React from "react"
import { render } from "@testing-library/react"
import { ListingDetailsLotteryPreferencesEducator } from "../../../modules/listingDetailsLottery/ListingDetailsLotteryPreferencesEducator"
import { lotteryResultRentalEducator } from "../../data/RailsLotteryResult/lottery-result-rental-educator"

describe("ListingDetailsLotteryPreferencesEducator", () => {
  it("displays 3 default preferences - COP, DTHP, L/W", () => {
    const { asFragment } = render(
      <ListingDetailsLotteryPreferencesEducator
        lotteryBucketsDetails={lotteryResultRentalEducator}
        isEducatorOne={false}
      />
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it("displays 2 preferences - NRHP, L/W", () => {
    const { asFragment } = render(
      <ListingDetailsLotteryPreferencesEducator
        lotteryBucketsDetails={lotteryResultRentalEducator}
        isEducatorOne={false}
      />
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it("educator 1 displays 3 default preferences - COP, DTHP, L/W", () => {
    const { asFragment } = render(
      <ListingDetailsLotteryPreferencesEducator
        lotteryBucketsDetails={lotteryResultRentalEducator}
        isEducatorOne={true}
      />
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it("educator 1 displays 2 preferences - NRHP, L/W", () => {
    const { asFragment } = render(
      <ListingDetailsLotteryPreferencesEducator
        lotteryBucketsDetails={lotteryResultRentalEducator}
        isEducatorOne={true}
      />
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
