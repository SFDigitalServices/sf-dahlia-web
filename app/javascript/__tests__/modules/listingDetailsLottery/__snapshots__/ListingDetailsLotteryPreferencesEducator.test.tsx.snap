// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingDetailsLotteryPreferencesEducator displays 2 preferences - NRHP, L/W 1`] = `
<DocumentFragment>
  <div
    class="text-sm"
  >
    <header
      class="border-b border-gray-450 mb-4"
    >
      <h2
        class="text__underline-weighted mx-8"
      >
        Priority of Housing Preferences
      </h2>
      <div
        class="pb-4 text-gray-700 text-xs mx-8 translate"
      >
        <p>
          Ranking in these lists is considered in the order shown here. 
          <a
            href="https://www.youtube.com/watch?v=oW56bUsrSW4"
            target="_blank"
          >
            Watch a video on how lottery ranking works.
          </a>
        </p>
      </div>
      <div
        class="pb-4 text-gray-700 text-xs mx-8"
      >
        US Military Veterans rank at the top of each category.
      </div>
    </header>
    <header
      class="border-b border-gray-450 mb-4"
    >
      <h2
        class="text__underline-weighted mx-8 text-xs"
      >
        Educators - San Francisco Unified School District
      </h2>
    </header>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Displaced Tenant Housing Preference (DTHP)
        </h3>
        <p
          class="text-sm"
        >
          Up to %{smart_count} unit |||| Up to %{smart_count} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 applicant is on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Neighborhood Resident Housing Preference (NRHP)
        </h3>
        <p
          class="text-sm"
        >
          Up to %{smart_count} unit |||| Up to %{smart_count} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 Veteran on this list.
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 other applicant on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Live or Work in San Francisco Preference
        </h3>
        <p
          class="text-sm"
        >
          Up to %{smart_count} unit |||| Up to %{smart_count} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          2 Veterans on this list.
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          3 other applicants on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <header
      class="border-b border-gray-450 mb-4"
    >
      <h2
        class="text__underline-weighted mx-8 text-xs"
      >
        All Other Employees - San Francisco Unified School District
      </h2>
    </header>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Certificate of Preference (COP)
        </h3>
        <p
          class="text-sm"
        >
          Up to %{smart_count} unit |||| Up to %{smart_count} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 applicant is on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Displaced Tenant Housing Preference (DTHP)
        </h3>
        <p
          class="text-sm"
        >
          Up to %{smart_count} unit |||| Up to %{smart_count} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 Veteran on this list.
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 other applicant on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Neighborhood Resident Housing Preference (NRHP)
        </h3>
        <p
          class="text-sm"
        >
          Up to %{smart_count} unit |||| Up to %{smart_count} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 applicant is on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Live or Work in San Francisco Preference
        </h3>
        <p
          class="text-sm"
        >
          Up to %{smart_count} unit |||| Up to %{smart_count} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 Veteran on this list.
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          2 other applicants on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <header
      class="border-b border-gray-450 mb-4"
    >
      <h2
        class="text__underline-weighted mx-8 text-xs"
      >
        General Public
      </h2>
    </header>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Certificate of Preference (COP)
        </h3>
        <p
          class="text-sm"
        >
          Up to 1 unit
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 applicant is on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Displaced Tenant Housing Preference (DTHP)
        </h3>
        <p
          class="text-sm"
        >
          Up to 1 unit
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          2 applicants are on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Neighborhood Resident Housing Preference (NRHP)
        </h3>
        <p
          class="text-sm"
        >
          Up to 1 unit
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          2 applicants are on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Live or Work in San Francisco Preference
        </h3>
        <p
          class="text-sm"
        >
          Up to 1 unit
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          5 applicants are on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          General Pool
        </h3>
        <p
          class="text-sm"
        >
          Up to 1 unit
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          2 applicants are on this list.
        </p>
      </div>
      <hr
        class="mt-4"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`ListingDetailsLotteryPreferencesEducator displays 3 default preferences - COP, DTHP, L/W 1`] = `
<DocumentFragment>
  <div
    class="text-sm"
  >
    <header
      class="border-b border-gray-450 mb-4"
    >
      <h2
        class="text__underline-weighted mx-8"
      >
        Priority of Housing Preferences
      </h2>
      <div
        class="pb-4 text-gray-700 text-xs mx-8 translate"
      >
        <p>
          Ranking in these lists is considered in the order shown here. 
          <a
            href="https://www.youtube.com/watch?v=oW56bUsrSW4"
            target="_blank"
          >
            Watch a video on how lottery ranking works.
          </a>
        </p>
      </div>
      <div
        class="pb-4 text-gray-700 text-xs mx-8"
      >
        US Military Veterans rank at the top of each category.
      </div>
    </header>
    <header
      class="border-b border-gray-450 mb-4"
    >
      <h2
        class="text__underline-weighted mx-8 text-xs"
      >
        Educators - San Francisco Unified School District
      </h2>
    </header>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Displaced Tenant Housing Preference (DTHP)
        </h3>
        <p
          class="text-sm"
        >
          Up to %{smart_count} unit |||| Up to %{smart_count} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 applicant is on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Neighborhood Resident Housing Preference (NRHP)
        </h3>
        <p
          class="text-sm"
        >
          Up to %{smart_count} unit |||| Up to %{smart_count} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 Veteran on this list.
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 other applicant on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Live or Work in San Francisco Preference
        </h3>
        <p
          class="text-sm"
        >
          Up to %{smart_count} unit |||| Up to %{smart_count} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          2 Veterans on this list.
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          3 other applicants on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <header
      class="border-b border-gray-450 mb-4"
    >
      <h2
        class="text__underline-weighted mx-8 text-xs"
      >
        All Other Employees - San Francisco Unified School District
      </h2>
    </header>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Certificate of Preference (COP)
        </h3>
        <p
          class="text-sm"
        >
          Up to %{smart_count} unit |||| Up to %{smart_count} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 applicant is on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Displaced Tenant Housing Preference (DTHP)
        </h3>
        <p
          class="text-sm"
        >
          Up to %{smart_count} unit |||| Up to %{smart_count} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 Veteran on this list.
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 other applicant on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Neighborhood Resident Housing Preference (NRHP)
        </h3>
        <p
          class="text-sm"
        >
          Up to %{smart_count} unit |||| Up to %{smart_count} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 applicant is on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Live or Work in San Francisco Preference
        </h3>
        <p
          class="text-sm"
        >
          Up to %{smart_count} unit |||| Up to %{smart_count} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 Veteran on this list.
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          2 other applicants on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <header
      class="border-b border-gray-450 mb-4"
    >
      <h2
        class="text__underline-weighted mx-8 text-xs"
      >
        General Public
      </h2>
    </header>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Certificate of Preference (COP)
        </h3>
        <p
          class="text-sm"
        >
          Up to 1 unit
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 applicant is on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Displaced Tenant Housing Preference (DTHP)
        </h3>
        <p
          class="text-sm"
        >
          Up to 1 unit
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          2 applicants are on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Neighborhood Resident Housing Preference (NRHP)
        </h3>
        <p
          class="text-sm"
        >
          Up to 1 unit
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          2 applicants are on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Live or Work in San Francisco Preference
        </h3>
        <p
          class="text-sm"
        >
          Up to 1 unit
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          5 applicants are on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          General Pool
        </h3>
        <p
          class="text-sm"
        >
          Up to 1 unit
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          2 applicants are on this list.
        </p>
      </div>
      <hr
        class="mt-4"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`ListingDetailsLotteryPreferencesEducator educator 1 displays 2 preferences - NRHP, L/W 1`] = `
<DocumentFragment>
  <div
    class="text-sm"
  >
    <header
      class="border-b border-gray-450 mb-4"
    >
      <h2
        class="text__underline-weighted mx-8"
      >
        Priority of Housing Preferences
      </h2>
      <div
        class="pb-4 text-gray-700 text-xs mx-8 translate"
      >
        <p>
          Ranking in these lists is considered in the order shown here. 
          <a
            href="https://www.youtube.com/watch?v=oW56bUsrSW4"
            target="_blank"
          >
            Watch a video on how lottery ranking works.
          </a>
        </p>
      </div>
      <div
        class="pb-4 text-gray-700 text-xs mx-8"
      >
        US Military Veterans rank at the top of each category.
      </div>
    </header>
    <header
      class="border-b border-gray-450 mb-4"
    >
      <h2
        class="text__underline-weighted mx-8 text-xs"
      >
        Educators - San Francisco Unified School District
      </h2>
    </header>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Displaced Tenant Housing Preference (DTHP)
        </h3>
        <p
          class="text-sm"
        >
          Up to %{smart_count} unit |||| Up to %{smart_count} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 applicant is on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Neighborhood Resident Housing Preference (NRHP)
        </h3>
        <p
          class="text-sm"
        >
          Up to %{smart_count} unit |||| Up to %{smart_count} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 Veteran on this list.
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 other applicant on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Live or Work in San Francisco Preference
        </h3>
        <p
          class="text-sm"
        >
          Up to %{smart_count} unit |||| Up to %{smart_count} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          2 Veterans on this list.
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          3 other applicants on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <header
      class="border-b border-gray-450 mb-4"
    >
      <h2
        class="text__underline-weighted mx-8 text-xs"
      >
        All Other Employees - San Francisco Unified School District
      </h2>
    </header>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Certificate of Preference (COP)
        </h3>
        <p
          class="text-sm"
        >
          Up to %{smart_count} unit |||| Up to %{smart_count} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 applicant is on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Displaced Tenant Housing Preference (DTHP)
        </h3>
        <p
          class="text-sm"
        >
          Up to %{smart_count} unit |||| Up to %{smart_count} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 Veteran on this list.
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 other applicant on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Neighborhood Resident Housing Preference (NRHP)
        </h3>
        <p
          class="text-sm"
        >
          Up to %{smart_count} unit |||| Up to %{smart_count} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 applicant is on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Live or Work in San Francisco Preference
        </h3>
        <p
          class="text-sm"
        >
          Up to %{smart_count} unit |||| Up to %{smart_count} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 Veteran on this list.
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          2 other applicants on this list.
        </p>
      </div>
      <hr
        class="mt-4"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`ListingDetailsLotteryPreferencesEducator educator 1 displays 3 default preferences - COP, DTHP, L/W 1`] = `
<DocumentFragment>
  <div
    class="text-sm"
  >
    <header
      class="border-b border-gray-450 mb-4"
    >
      <h2
        class="text__underline-weighted mx-8"
      >
        Priority of Housing Preferences
      </h2>
      <div
        class="pb-4 text-gray-700 text-xs mx-8 translate"
      >
        <p>
          Ranking in these lists is considered in the order shown here. 
          <a
            href="https://www.youtube.com/watch?v=oW56bUsrSW4"
            target="_blank"
          >
            Watch a video on how lottery ranking works.
          </a>
        </p>
      </div>
      <div
        class="pb-4 text-gray-700 text-xs mx-8"
      >
        US Military Veterans rank at the top of each category.
      </div>
    </header>
    <header
      class="border-b border-gray-450 mb-4"
    >
      <h2
        class="text__underline-weighted mx-8 text-xs"
      >
        Educators - San Francisco Unified School District
      </h2>
    </header>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Displaced Tenant Housing Preference (DTHP)
        </h3>
        <p
          class="text-sm"
        >
          Up to %{smart_count} unit |||| Up to %{smart_count} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 applicant is on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Neighborhood Resident Housing Preference (NRHP)
        </h3>
        <p
          class="text-sm"
        >
          Up to %{smart_count} unit |||| Up to %{smart_count} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 Veteran on this list.
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 other applicant on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Live or Work in San Francisco Preference
        </h3>
        <p
          class="text-sm"
        >
          Up to %{smart_count} unit |||| Up to %{smart_count} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          2 Veterans on this list.
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          3 other applicants on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <header
      class="border-b border-gray-450 mb-4"
    >
      <h2
        class="text__underline-weighted mx-8 text-xs"
      >
        All Other Employees - San Francisco Unified School District
      </h2>
    </header>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Certificate of Preference (COP)
        </h3>
        <p
          class="text-sm"
        >
          Up to %{smart_count} unit |||| Up to %{smart_count} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 applicant is on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Displaced Tenant Housing Preference (DTHP)
        </h3>
        <p
          class="text-sm"
        >
          Up to %{smart_count} unit |||| Up to %{smart_count} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 Veteran on this list.
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 other applicant on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Neighborhood Resident Housing Preference (NRHP)
        </h3>
        <p
          class="text-sm"
        >
          Up to %{smart_count} unit |||| Up to %{smart_count} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 applicant is on this list.
        </p>
      </div>
      <hr
        class="my-4"
      />
    </div>
    <div>
      <div
        class="px-8"
      >
        <h3
          class="font-sans font-semibold text-xs tracking-wide uppercase"
        >
          Live or Work in San Francisco Preference
        </h3>
        <p
          class="text-sm"
        >
          Up to %{smart_count} unit |||| Up to %{smart_count} units
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          1 Veteran on this list.
        </p>
        <p
          class="text-gray-700 text-sm"
        >
          2 other applicants on this list.
        </p>
      </div>
      <hr
        class="mt-4"
      />
    </div>
  </div>
</DocumentFragment>
`;
